src/sys/ptrace.rs

```diff
@@ -275,3 +280,209 @@ pub fn cont<T: Into<Option<Signal>>>(pid: Pid, sig: T) -> Result<()> {
     }
 }
 
+/// Represents all possible ptrace-accessible registers on x86_64
+#[cfg(target_arch = "x86_64")]
+#[allow(non_camel_case_types)]
+#[derive(Debug, PartialEq)]
+pub enum Register {
+    R15 = 8 * ::libc::R15 as isize,
+    R14 = 8 * ::libc::R14 as isize,
+    R13 = 8 * ::libc::R13 as isize,
+    R12 = 8 * ::libc::R12 as isize,
+    RBP = 8 * ::libc::RBP as isize,
+    RBX = 8 * ::libc::RBX as isize,
+    R11 = 8 * ::libc::R11 as isize,
+    R10 = 8 * ::libc::R10 as isize,
+    R9 = 8 * ::libc::R9 as isize,
+    R8 = 8 * ::libc::R8 as isize,
+    RAX = 8 * ::libc::RAX as isize,
+    RCX = 8 * ::libc::RCX as isize,
+    RDX = 8 * ::libc::RDX as isize,
+    RSI = 8 * ::libc::RSI as isize,
+    RDI = 8 * ::libc::RDI as isize,
```

Comment:
I think we can definitely document these registers. For example `RDI` is the "Destination Index register. Used as a pointer to a destination in stream operations" (taken from [here](https://en.wikibooks.org/wiki/X86_Assembly/X86_Architecture)). Even for R15 you can write "General purpose register 15", which is more descriptive than the name itself.

I think it's important to be kind to our users by a) not assuming too much about their knowledge and b) to lead them in the right direction if they want to know more.

Reply:
`General purpose register 15` says nothing. You need to know what the ABI is to do anything with ptrace. And if they need to search, they'll be better off searching for the 3-letter names, since it's what the whole C API uses and what everything uses.

Being kind is one thing, but we can't teach them everything. There's some learning curve for using ptrace and knowing registers is just a thing. The process has to be stopped, or weird errors will arise. It's out of scope for nix to provide a ptrace tutorial.

The thing about learning curve is that it protects users who don't know enough from shooting themselves in the foot by using something they don't understand.

If you insist, we can add a link to the wikibook in the docs.

Reply:
If you feel so strongly about this, then let's drop this as it's mostly a nice-to-have. Eventually I'm going to enable a requirement that everything have docs, but since we're not there yet this isn't too important.